src/core_atmosphere/dynamics/mpas_atm_time_integration.F

```diff
@@ -7474,55 +7553,93 @@ subroutine summarize_timestep(domain)
          !
          ! Check for NaNs
          !
+         found_NaN = .false.
+
+         !$acc parallel default(present)
+         !$acc loop collapse(2) gang vector reduction(.or.:found_NaN)
          do iCell = 1, nCellsSolve
          do k = 1, nVertLevels
             if (ieee_is_nan(w(k,iCell))) then
-               call mpas_log_write('NaN detected in ''w'' field.', messageType=MPAS_LOG_CRIT)
+               found_NaN = .true.
             end if
          end do
          end do
+         !$acc end parallel
+         if (found_NaN) then
+            call mpas_log_write('NaN detected in ''w'' field.', messageType=MPAS_LOG_CRIT)
+         end if
 
+
```

Comment:
If we reach this line with the code as written, `found_NaN` is guaranteed to be `.false.`. However, if we were ever to change the log messages from critical error messages (that halt execution) to regular error messages, we may then get false messages about NaNs in the `u` field below. To improve robustness of the code, should we reset `found_NaN` to `.false.` here?